### PR TITLE
feat: Add `PrivateDirectory::entires`, `PrivateFile::read_at` and make `PrivateFile::get_content_size_upper_bound` public

### PR DESCRIPTION
**Pull request by Frando**
This PR contains changes I needed to made while hacking on [wnfs-fuse](https://github.com/Frando/wnfs-fuse) at IPFSThing.

- Public getter method on `PrivateDirectory` to get the entries. `ls` exists, but loads all the metadata, which is expensive if not needed
- Make `PrivateFile::get_content_size_upper_bound` public (to have something to report as filesize - even if inaccurate). It might be a good idea to add the actual filesize to the metadata instead.
- Add a new method `PrivateFile::read_at` to read `size` bytes from `offset`. This makes use of `stream_content` internally and does the block and offset calculations

### CODE REVIEW

**Comment by appcypher** [comment]
Amazing work on `wnfs-fuse` and thanks for the contribution.
This looks great to me. Just minor nits.

File: wnfs/src/private/directory.rs
Comment (appcypher) [inline wnfs/src/private/directory.rs:1007]: Can you add a tiny doc to the function? 
Nit: Can you also rename the function `get_entries` just to be consistent with the rest of the code?

File: wnfs/src/private/file.rs
Comment (appcypher) [inline wnfs/src/private/file.rs:552]: 👍🏾

File: wnfs/src/private/directory.rs
Comment (Frando) [reply wnfs/src/private/directory.rs]: :+1:

**Comment by Frando** [comment]
I also added a test for `PrivateFile::read_at`. Don't have much experience with proptests, but copied one of the other tests and adapted it.

File: wnfs/src/private/file.rs
Comment (matheus23) [inline wnfs/src/private/file.rs:384]: Can you make this just error out?
We shouldn't return any bytes if the content is corrupted. Usually we'd do `InvalidSerialization` or w/e it was called, but adding a custom error type may not be bad either.

**Review by matheus23 — APPROVED** [review]
One critical change about erroring out required, but LGTM when that's handled :) :rocket:

File: wnfs/src/private/file.rs
Comment (Frando) [reply wnfs/src/private/file.rs]: Yup, makes sense.

**Pull request merged**
